batch/batch/batch.py

```diff
@@ -517,6 +546,15 @@ def is_successful(self):
             pod_status = pod.status.to_str()
             terminated = pod.status.container_statuses[0].state.terminated
             exit_code = terminated.exit_code
+
+            if task_name == 'main':
+                init_terminated = pod.status.init_container_statuses[0].state.terminated
+                init_exit_code = init_terminated.exit_code
+                if init_exit_code == 1:
+                    log.info(f"{self.full_id} was previously run; restarting job")
+                    await self.reset()
+                    return
```

Comment:
So this happens if a rival batch instance creates a second main pod. If the previous main pod finished successfully, I don't think there's any problem. I think we should hit the db and check if we have an exit code for main already. If yes, do nothing. If no, then do the above.

Reply:
We probably need to delete this pod, right? Otherwise we will fail to create a main pod later after we've reset.

Reply:
That occurs in reset -> mark_unscheduled -> delete_pod